server/__tests__/integration/competition.test.js

```diff
@@ -0,0 +1,512 @@
+        startsAt: '2025-05-17T22:00:00.000Z',
+        endsAt: '2025-05-17T22:00:00.000Z'
+      });
+
+      expect(response.status).toBe(201);
+      expect(response.body.title).toMatch('test');
+      expect(response.body.metric).toMatch('overall');
+      expect(response.body.startsAt).toMatch('2025-05-17T22:00:00.000Z');
+      expect(response.body.endsAt).toMatch('2025-05-17T22:00:00.000Z');
+
+      TEST_DATA.minimal = response.body;
+
+      done();
+    });
+
+    test('Create with group', async done => {
+      const response = await request.post('/api/competitions').send({
+        title: 'test',
+        metric: 'overall',
+        startsAt: '2025-05-17T22:00:00.000Z',
+        endsAt: '2025-05-17T22:00:00.000Z',
+        groupId: TEST_GROUP_ID,
+        groupVerificationCode: TEST_GROUP_VERIFICATION_CODE
+      });
+
+      expect(response.status).toBe(201);
+      expect(response.body.participants.length).toBe(2);
+
+      TEST_DATA.group = response.body;
+
+      done();
+    });
+
+    test('Do not create ( invalid group Id )', async done => {
+      const response = await request.post('/api/competitions').send({
+        title: 'test',
+        metric: 'overall',
+        startsAt: '2025-05-17T22:00:00.000Z',
+        endsAt: '2025-05-17T22:00:00.000Z',
+        groupId: 1337,
+        groupVerificationCode: TEST_GROUP_VERIFICATION_CODE
+      });
+
+      expect(response.status).toBe(400);
+      expect(response.body.message).toMatch('Invalid group id.');
+
+      done();
+    });
+
+    test('Do not create ( no group verification )', async done => {
+      const response = await request.post('/api/competitions').send({
+        title: 'test',
+        metric: 'overall',
+        startsAt: '2025-05-17T22:00:00.000Z',
+        endsAt: '2025-05-17T22:00:00.000Z',
+        groupId: TEST_GROUP_ID
+      });
+
+      expect(response.status).toBe(400);
+      expect(response.body.message).toMatch('Invalid verification code.');
+
+      done();
+    });
+
+    test('Do not create (invalid participant type data)', async done => {
+      const response = await request.post('/api/competitions').send({
+        title: 'test',
+        metric: 'overall',
+        startsAt: '2025-05-17T22:00:00.000Z',
+        endsAt: '2025-05-25T22:00:00.000Z',
+        participants: [{}, 123, 'valid']
+      });
+
+      expect(response.status).toBe(400);
+      expect(response.body.message).toContain(
+        '2 Invalid usernames: Names must be 1-12 characters long,'
+      );
+
+      done();
+    });
+  });
+
+  describe('Viewing', () => {
+    test('List competitions', async done => {
+      const response = await request.get(`/api/competitions`).send();
+
+      expect(response.status).toBe(200);
+      expect(response.body.length).toBe(3);
+
+      done();
+    });
+
+    test('Search competitions ( invalid status )', async done => {
+      const response = await request.get(`/api/competitions?status=invalid-status`).send();
```

Comment:
It would be better to not hardcode the query params, check the Player search tests to see how you can do it differently ;)